contrib/dockerfile_build.bzl

```diff
@@ -0,0 +1,65 @@
+# Copyright 2016 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Repository rule to build image with `docker build`."""
+
+def _impl(rctx):
+  tag = rctx.attr.tag
+  dockerfile = rctx.path(rctx.attr.dockerfile)
+  result = rctx.execute([
+      "docker",
```

Comment:
Docker builds can run into trouble with build contexts that contain symlinks. In the past when doing things like this I've had to "resolve" symlinks by copying to a tempdir before doing doing the docker build. Does this cause any trouble here?